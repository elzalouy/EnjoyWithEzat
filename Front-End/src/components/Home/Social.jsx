import { Fragment } from "react";
import { useSelector } from "react-redux";

const Social = () => {
  // I'm here calling the store from redux to get the data i need. It could be editable someday.
  const social = useSelector((state) => state.Person.social);
  return (
    <Fragment>
      {social.map((item) => (
        <a key={item.name} href={item.link} target="blank">
          <i className={"col-1 p-2 bi " + item.icon}></i>
        </a>
      ))}
    </Fragment>
  );
};

export default Social;