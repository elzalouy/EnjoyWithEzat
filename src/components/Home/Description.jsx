import { Fragment } from "react";
import { useSelector } from "react-redux";
const Description = () => {
  // from the store i get my data
    const data=useSelector(state=>state.Person);
  return (
    <Fragment>
      <h1>{data.name}</h1>
      <h5>{data.jobTitle}</h5>
    </Fragment>
  );
};

export default Description;