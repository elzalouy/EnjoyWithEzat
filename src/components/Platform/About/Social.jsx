import { useSelector } from "react-redux";
import classes from "./index.module.css";
const SocialItems = () => {
  const socialItems = useSelector((state) => state.Person.social);
  return (
      <div className={classes.footer + "  bg-3"}>
        {socialItems.map((item) => (
          <a
            key={item.name}
            href={item.link}
            className={"Btn m-0 p-0"}
            target="blank"
          >
            <i className={  "Btn  gray p-2 m-0 bi " + item.icon}></i>
          </a>
        ))}
      </div>
  );
};

export default SocialItems;
