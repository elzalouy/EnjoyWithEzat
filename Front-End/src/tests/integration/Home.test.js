import Description from '../../components/Home/Description';
import Social from '../../components/Home/Social';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createMockStore from 'redux-mock-store';
import reduxStore from '../../store';
const mockstore = createMockStore([]);
describe('Home > Description Section : ', () => {
    let store = mockstore({ Person: { name: 'Ezat Elzalouy', jobTitle: 'MERN stack developer' } })
    let component = renderer.create(<Provider store={store}><Description /></Provider>)
    test('component should be rendered with a state from redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })
});

describe('Home > Social Section', () => {
    let store = mockstore({
        Person: reduxStore.getState().Person
    })
    let component=renderer.create(<Provider store={store}><Social/></Provider>)
    test('component should be rendered with a state from redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })
})