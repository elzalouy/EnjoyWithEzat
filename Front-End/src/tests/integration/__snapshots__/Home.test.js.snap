// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home > Description Section :  component should be rendered with a state from redux store 1`] = `
Array [
  <h4
    className="bolder-3"
  >
    Ezat Elzalouy
  </h4>,
  <h5>
    MERN stack developer
  </h5>,
]
`;

exports[`Home > Social Section component should be rendered with a state from redux store 1`] = `
Array [
  <a
    href="https://www.facebook.com/Eng.ezat.elzalouy/"
    target="blank"
  >
    <i
      className="col-1 p-2 bi bi-facebook"
    />
  </a>,
  <a
    href="https://twitter.com/Ezat_Elzalouy/"
    target="blank"
  >
    <i
      className="col-1 p-2 bi bi-twitter"
    />
  </a>,
  <a
    href="https://www.linkedin.com/in/ezat-elzalouy-844bb0133/"
    target="blank"
  >
    <i
      className="col-1 p-2 bi bi-linkedin"
    />
  </a>,
  <a
    href="https://github.com/elzalouy"
    target="blank"
  >
    <i
      className="col-1 p-2 bi bi-github"
    />
  </a>,
]
`;
